docs/source/topics/data_types.rst

```diff
@@ -4,8 +4,790 @@
+
+  In [1]: FolderData = DataFactory('folder')
+
+  In [2]: folder = FolderData(tree='/absolute/path/to/directory')
+
+Alternatively, you can construct the node first and then use the various repository methods to add objects from directory and file paths:
+
+.. code-block:: ipython
+
+  In [1]: folder = FolderData()
+
+  In [2]: folder.put_object_from_tree('/absolute/path/to/directory')
+
+  In [3]: folder.put_object_from_file('/absolute/path/to/file1.txt', path='file1.txt')
+
+or from `file-like objects <https://docs.python.org/3/glossary.html#term-file-like-object>`_:
+
+.. code-block:: ipython
+
+  In [4]: folder.put_object_from_filelike(filelike_object, path='file2.txt')
+
+Inversely, the content of the files stored in the :py:class:`~aiida.orm.nodes.data.folder.FolderData` node can be accessed using the :py:class:`~aiida.orm.nodes.node.Node.get_object_content()` method:
+
+.. code-block:: ipython
+
+  In [5]: folder.get_object_content('file1.txt')
+  Out[5]: 'File 1 content\n'
+
+To see the files that are stored in the :py:class:`~aiida.orm.nodes.data.folder.FolderData`, you can use the :py:class:`~aiida.orm.nodes.node.Node.list_object_names()` method:
+
+.. code-block:: ipython
+
+  In [6]: folder.list_object_names()
+  Out[6]: ['subdir', 'file1.txt', 'file2.txt']
+
+In this example, ``subdir`` was a sub directory of ``/absolute/path/to/directory``, whose contents where added above.
+to list the contents of the ``subdir`` directory, you can pass its path to the :py:meth:`~aiida.orm.nodes.node.Node.list_object_names()` method:
+
+.. code-block:: ipython
+
+  In [7]: folder.list_object_names('subdir')
+  Out[7]: ['file3.txt', 'module.py']
+
+The content can once again be shown using the :py:meth:`~aiida.orm.nodes.node.Node.get_object_content()` method by passing the correct path:
+
+.. code-block:: ipython
+
+ In [8]: folder.get_object_content('subdir/file3.txt')
+ Out[8]: 'File 3 content\n'
+
+Since the :py:class:`~aiida.orm.nodes.data.folder.FolderData` node is simply a collection of files, it simply stores these files in the repository
+
+.. _topics:data_types:core:remote:
+
+RemoteData
+----------
+
+The :py:class:`~aiida.orm.nodes.data.remote.RemoteData` node represents a "symbolic link" to a specific folder on a remote computer.
+Its main use is to allow users to persist the provenance when e.g. a calculation produces data in a raw/scratch folder, and the whole folder needs to be provided to restart/continue.
+To create a :py:class:`~aiida.orm.nodes.data.remote.RemoteData` instance, simply pass the remote path to the folder and the computer on which it is stored:
+
+.. code-block:: ipython
+
+  In [1]: RemoteData = DataFactory('remote')
+
+  In [2]: computer = load_computer(label='computer_label')
+
+  In [3]: remote = RemoteData(remote_path='/absolute/path/to/remote/directory' computer=local)
+
+You can see the contents of the remote folder by using the :py:meth:`~aiida.orm.nodes.data.remote.RemoteData.listdir()` method:
```

Comment:
A big warning is due here - this will open a SSH connection, so this should NOT be done in scripts and even less in workflows; it's just a utility for the user.